src/plugins/mediawiki.lua

```diff
@@ -7,31 +7,41 @@ local plugin = {}
 plugin.help = 'Usage: mediawiki <APIURL> <search>'
 
 plugin.main = function(args)
-    local _, _, apiurl, search = args.message:find('(%S+)%s+(.+)')
+    local _, _, apiurl, search = args.message:find('^[ ]?(https?://[^> ]+)%s+(.+)')
 
     if args.modules.config.debug then
         print(apiurl, query)
     end
 
     if not apiurl then
-        args.modules.irc.privmsg(args.target, 'You want me to search where?')
+        args.modules.irc.privmsg(args.target, ('%s: Please give me a mediawiki API url, it should have api.php somewhere on the end.'):format(args.sender))
+        return
     end
 
     if not search then
-        args.modules.irc.privmsg(args.target, 'You want me to search for what?')
+        args.modules.irc.privmsg(args.target, ('%s: Please give me a search term.'):format(args.sender))
+        return
     end
 
     local act = '?action=opensearch&format=json&search='
     local resp = https.request(apiurl .. act .. url.escape(search))
-    if resp and resp not nil then
-        if args.modules.config.debug then print(resp) end
-        local res = json.decode(resp)
-        local lnk = (res[4][1] and res[4][1] ~= '') and res[4][1] or 'No results'
-        local dsc = (res[3][1] and res[3][1] ~= '') and ' - ' .. res[3][1] or ''
-        args.modules.irc.privmsg(args.target, ('<%s>%s'):format(lnk, dsc))
+    if not resp or resp == nil then
```

Comment:
This is redundant. `not resp` is shorthand for `resp == nil`. It was there before (because I am either crazy or missed it).

Reply:
also fixed \o/